### ISSUE TITLE
[HOLD for payment 2025-01-02] [$500] iOS Invoices - App freezes after deleting bank account

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#52932]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.65-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**Email or phone of affected tester (no customers):** applausetester+zm57@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Precondition:
- User has added bank account in Invoices section.

1. Launch ND or hybrid app.
2. Go to workspace settings > Invoices.
3. Tap on the bank account.
4. Tap Delete.
5. Delete the bank account.
6. Tap anywhere else.

## Expected Result:
App will not freeze after deleting bank account.

## Actual Result:
App freezes after deleting bank account.

## Workaround:
Unknown

## Platforms:

- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/26cd56e5-5a3e-4118-9cef-daab03db81db

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021861553347426346949</li>
        <li>Upwork Job ID: 1861553347426346949</li>
        <li>Last Price Increase: 2024-12-17</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Reviewer | 105370190</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @twisterdotcom</details>

**Comment by twisterdotcom** [comment]
Wow, it absolutely does.

**Comment by twisterdotcom** [comment]
iOS:
https://github.com/user-attachments/assets/5213f865-66e1-4c29-b7d4-fa43123c7c73

**Comment by twisterdotcom** [comment]
Waiting on some proposals

**Comment by DylanDylann** [comment]
Waiting for proposals

**Comment by twisterdotcom** [comment]
@yuwenmemon have you had a chance to look at this yet at all?

**Comment by yuwenmemon** [comment]
Hmmm... no - should we open a thread in #expensify-open-source?

**Comment by muttmuure** [comment]
@DylanDylann can you clarify if this is internal or external?

**Comment by twisterdotcom** [comment]
I have to imagine it's internal given the crash. All of the request IDs in the logs above don't really give us much either:

```
8e8da90898acef0f-LHR
8e8da69f0fb9ecfe-LHR
8e8da697bcfaecfe-LHR
8e8da6735f2cecfe-LHR
8e8da6692c02ecfe-LHR
```

They're not for deleting a bank account. It's like the app completely froze when doing that. Will see if we can get a retest for now.

**Comment by twisterdotcom** [comment]
I have to imagine it's internal given the crash. All of the request IDs in the logs above don't really give us much either:

```
8e8da90898acef0f-LHR
8e8da69f0fb9ecfe-LHR
8e8da697bcfaecfe-LHR
8e8da6735f2cecfe-LHR
8e8da6692c02ecfe-LHR
```

They're not for deleting a bank account. It's like the app completely froze when doing that. Will see if we can get a retest for now.

**Comment by DylanDylann** [comment]
It is a front-end bug and should be external

**Comment by DylanDylann** [comment]
Still waiting for proposals

**Comment by muttmuure** [comment]
If it's external let's keep increasing the price

**Comment by koko57** [comment]
Hello, I'm Agata from Callstack, maybe I can help with this issue?

**Comment by koko57** [comment]
I remember I was working on a very similar issue - as two modals are used here something is preventing the one of the modals backdrop host components to be properly removed and as it's transparent it looks like the app is frozen (but it's not, just the transparent backdrop prevents from interacting with the components under). 
I've also found another issue - https://github.com/Expensify/App/issues/48298 - that seems to have the same root cause and the [solution](https://github.com/Expensify/App/pull/48332/files) used there should work here as well (probably with a slight modification, because it has changed a bit since the solution was implemented)

https://github.com/allgandalf/App/blob/d86024e6688326b5c3040a3c752e0b00265389de/src/pages/settings/Wallet/WalletPage/WalletPage.tsx#L598-L606

**Comment by koko57** [comment]
That additional change was made because of the restriction for copilot. Should the copilot be able to delete the bank account from the invoices?

**Comment by koko57** [comment]
ok, thanks! So it means that just this one line is fine. I tested this solution and it works fine, so I will open a PR for that

**Comment by DylanDylann** [comment]
@koko57 Yeah, feel free to open a PR

**Comment by koko57** [comment]
@DylanDylann PR opened https://github.com/Expensify/App/pull/54279

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by twisterdotcom** [comment]
Payment Summary:

- @koko57 **does not** require payment (Contractor)
- @DylanDylann paid $500 [here](https://www.upwork.com/nx/wm/offer/105370190) (Reviewer)
